bin/syclcc

```diff
@@ -447,19 +501,33 @@ class syclcc_config:
         print(""" 
 --force-alternative-compiler=<compiler>  Use the specified compiler
                directly without going through the hipSYCL toolchain
+--cuda-clang-compiler=<compiler>  Use an alternative version of Clang
+               for compiling CUDA (defaults to the newest version of
+               clang++ that can be found)
 --keep-temporary-files Keep temporary files created by the hipSYCL
                source-to-source transformation; this can be useful
                for debugging
 --hipsycl-bootstrap  Enter bootstrap mode; do not perform source-to-
                source transformation. This is normally only required
                when building hipSYCL itself.
 --hipsycl-platform=<platform>  Build the program for the specified
-               hipSYCL platform; valid values are: cuda,rocm,cpu
+               hipSYCL platform; valid values are: cuda,nvcc,rocm,cpu
 --help Print help message """)
         sys.exit(0)
       else:
         self._forwarded_args.append(arg)
 
+    if self._cuda_clang_compiler == None:
+      self._cuda_clang_compiler = get_first_working_command([
+      "clang++-8.0","clang++-8","clang++-7.0","clang++-7",
+      "clang++-6.0","clang++-6","clang++"])
```

Comment:
Can we reuse this list in `get_host_compiler()`? Then we only have to maintain one list of clang executable names ;)

Reply:
Sure, however would you still like to default to `g++`?

Reply:
`get_host_compiler()` is only used for the CPU backend (which does almost not depend on the source-to-source transformation at all), so there's no inherent advantage to preferring clang. And since g++ comes preinstalled basically everywhere, I think it makes sense to use it as default. In the future we could however perhaps add a configuration flag for this.